### PR TITLE
chore: install dependencies directly in package

### PR DESCRIPTION
**Pull request by jviotti**
Previously, the build scripts would override the top level
`node_modules` and `bower_components`. After this commit, the
dependencies are installed directly in the Electron package.

Signed-off-by: Juan Cruz Viotti <jviotti@openmailbox.org>

### CODE REVIEW

**Comment by lurch** [comment]
Would it also make sense to include the OS and ARCH (and perhaps also GUI / CLI) in the prefix, so that we can have multiple versions installed side-by-side?

**Comment by jviotti** [comment]
Yeah, definitely. I'm doing so on the final Makefile (to not do work in vain on something that will be killed very soon). This is more or less what I have locally:

```
release/electron-$(PLATFORM)-$(ARCH)-app:
	./scripts/unix/create-electron-app.sh -s . -f "$(APPLICATION_FILES)" -o $@

release/electron-$(PLATFORM)-$(ARCH)-app/node_modules:\
	release/electron-$(PLATFORM)-$(ARCH)-app
	./scripts/unix/dependencies.sh -p -r "$(ARCH)" -v "$(ELECTRON_VERSION)" -x $< -t electron

release/electron-$(PLATFORM)-$(ARCH)-app.asar: \
	release/electron-$(PLATFORM)-$(ARCH)-app \
	release/electron-$(PLATFORM)-$(ARCH)-app/node_modules
	./scripts/unix/create-asar.sh -d $< -o $@
```

File: scripts/unix/dependencies.sh [hunk 1]
```diff
@@ -96,9 +97,25 @@ if [ "$ARGV_PRODUCTION" == "true" ]; then
...
 fi
 
+if [ -n "$ARGV_PREFIX" ]; then
+  NPM_INSTALL_OPTS="$NPM_INSTALL_OPTS --prefix=$ARGV_PREFIX"
+fi
+
 npm install $NPM_INSTALL_OPTS
 
+# Using `--prefix` might cause npm to create an empty `etc` directory
+if [ -n "$ARGV_PREFIX" ] && [ ! "$(ls -A "$ARGV_PREFIX/etc")" ]; then
```
Comment (lurch) [inline scripts/unix/dependencies.sh]: Why the use of `$(ls -A ...)` here? Wouldn't a simple `-d` test suffice?

Comment (lurch) [inline scripts/unix/dependencies.sh]: Ahhh, that makes more sense then.

**Pull request merged**